docs/DeveloperGuide.md

```diff
@@ -1,38 +1,314 @@
+
+#### Design considerations for Timer class
+- Current implementation: Create `Timer` from scratch, using the sleep function of threads to keep 
+track of time
+  - Pros: Same overhead of needing to track the time left of the timer
+  - Pros: No need to follow Java’s `Timer` class syntax, which can be confusing at times
+  - Pros: Implementation is simple and straight-forward
+  - Cons: Have to manage how we interrupt the thread after stopping the timer
+- Alternative: Using Java's `Timer` class
+  - Pros: The way of keeping track of the time has already been implemented
+  - Pros: Using a standard library usually makes the program less prone to various errors
+  - Cons: Still have to implement a way to keep track of time for our purposes of pausing a timer, since the library 
+  provided by Java has no way of pausing the timer, only stopping it.
+
+
+### Timetable implementation 
+
+The **TimeTable** component prints the daily or weekly schedule that the user wishes to see. 
+
+The functionalities of the timetable include:
+- Prints a schedule specific to the date the user inputs 
+- Prints the schedule for the week the user is at
+- Prints the schedule of the day whenever the user starts up Sherpass.
+- The timetable schedule is represented in a table form as shown below:
+
+![](images/timetableFormat.png)
+
+#### Task index of a task in timetable
+
+The task number in the timetable as shown follows the index of a task
+in the overall task list, i.e. the list containing all tasks added/edited. 
+This allows a more intuitive approach towards adding/editing/deleting/marking/unmarking of tasks.
+
+#### Time and Day column in timetable
+
+The **Time** and **Day** in the timetable follows the doOnDate attribute of a task.
+Concept wise, this treats the doOnDate as the date and time when the task occurs,
+or the date and time the user has set out to accomplish the task.
+
+
+The **Timetable** is a class which interacts with the following components:
+1. Parser
+2. ShowCommand
+3. TaskList
+
+#### Parser Component
+
+The **Parser** is a class which parses the inputs which the user enters. 
+To activate the timetable, the user inputs commands that start with `show`.
+This creates a **ShowCommand** object which will execute its method, thereby printing the timetable.
+
+#### ShowCommand Component
+
+Depending on the user input that was parsed by **Parser**, **ShowCommand** will call the 
+relative methods which prints the timetable.
+
+#### TaskList Component
+
+As **ShowCommand** is being executed, it will retrieve a filtered list
+of task by the date that is defined in the **ShowCommand** from the **TaskList** component. 
+The filtered list represents the schedule that the user has on that given date. 
+The list is assumed to be sorted previously when the user added/edited a task.
+
+Below is a sequence diagram of what happens 
+as the user wishes to see the schedule (timetable) for 25th May 2022:
+
+![](images/showScheduleForADate.png)
```

Comment:
I agree with sean, diagram is rather complicated and I think it would be better to use a reference frame and split this diagram into multiple diagrams to see each one clearer.

Reply:
Following up from my comment of the previous diagram, the Main class seems to be reading the userInput?